---
title: "Proteomic ruler:  Estimate absolute protein quantification accuracy"
author: "Cox Lab"
format: 
  html:
    toc: true
    toc-depth: 4
    toc-expand: false
    number-sections: true
    number-depth: 4
editor: source
date: today
bibliography: references.bib
csl: nature.csl
---

If you arrived here directly, it is a good idea to read the [Proteomic ruler overview](docs/proteomicruler.html) first.

# Description

Once you [estimated copy numbers and concentrations](docs/estimatecopynumbers.html), it is often desirable to get a feeling for the accuracy of the estimations. While exact error estimates are conceptually tricky because of the 'protease bias' (see [Peng, M. et al.](http:_www.ncbi.nlm.nih.gov/pubmed/22669647) [@peng2012]), there are some empirical rules that one can apply.

As a rule of thumb, quantification accuracy will increase with:

* the number of observed peptides
* the fraction of the protein sequence that is coverable (i.e. theoretical peptides/sequence length)
* the fraction of razor + unique peptides (which the quantification in based on) to total peptides. Only if most of the peptides of a protein were actually counted, one can expect accurate quantification.

# Parameters

##  Total number of peptides Select 

The _Peptides_ column of the MaxQuant output table. 

## Unique + razor peptides 

Select the _Unique + razor peptides_ column of the MaxQuant output table. 

## Sequence length 

Select either the _Sequence length_ column of the MaxQuant output table or the corresponding annotation column generated by [Annotate proteins](docs/annotateproteins.html). 

## Number of theoretical peptides 

Select the corresponding annotation column generated by [Annotate proteins](docs/annotateproteins.html). 

## Confidence class thresholds 

For the _high_ and _medium_ confidence class, specify the following thresholds. Every protein that does not fulfill the criteria for _high_ or _medium_ accuracy will be classified as _low_ accuracy. 

* **Min. peptides**: The minimum number of detected peptides. 
* **Min. razor fraction**: The minimum ratio of razor+unique/total peptides. 
* **Min. theoretical peptides** per 100 amino acids.

# Output

Your output table will contain an additional categorical column with the quantification accuracy confidence classes for each protein. In addition to looking at the confidence class of your protein of interest, you can also use [Category counting](docs/categorycounting.html) to look at the global distribution of confidence classes.
