<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.294">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Cox Lab">
<meta name="dcterms.date" content="2023-11-03">

<title>Activities</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    <img src="./Figures/coxlab_logo2.png" alt="" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html" rel="" target="">
 <span class="menu-text">HOME</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./maxquant_instructions.html" rel="" target="">
 <span class="menu-text">MaxQuant</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./perseus_instructions.html" rel="" target="">
 <span class="menu-text">Perseus</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./andromeda_instructions.html" rel="" target="">
 <span class="menu-text">Andromeda</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="./contact.html" rel="" target=""><i class="bi bi-envelope" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/cox-labs/CoxLab_Bug_Reporting" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/TheCoxLab" rel="" target=""><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/juergen-cox-2437ba17/" rel="" target=""><i class="bi bi-fa-linkedin fa-lg" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="-1">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#matrix-upload" id="toc-matrix-upload" class="nav-link active" data-scroll-target="#matrix-upload"><span class="header-section-number">1</span> Matrix Upload</a></li>
  <li><a href="#matrix-export" id="toc-matrix-export" class="nav-link" data-scroll-target="#matrix-export"><span class="header-section-number">2</span> Matrix Export</a></li>
  <li><a href="#matrix-analysis" id="toc-matrix-analysis" class="nav-link" data-scroll-target="#matrix-analysis"><span class="header-section-number">3</span> Matrix Analysis</a>
  <ul>
  <li><a href="#clusteringpca" id="toc-clusteringpca" class="nav-link" data-scroll-target="#clusteringpca"><span class="header-section-number">3.1</span> Clustering/PCA</a></li>
  <li><a href="#misc.-analysis" id="toc-misc.-analysis" class="nav-link" data-scroll-target="#misc.-analysis"><span class="header-section-number">3.2</span> Misc. (Analysis)</a></li>
  <li><a href="#visualization" id="toc-visualization" class="nav-link" data-scroll-target="#visualization"><span class="header-section-number">3.3</span> Visualization</a></li>
  </ul></li>
  <li><a href="#matrix-multiprocessing" id="toc-matrix-multiprocessing" class="nav-link" data-scroll-target="#matrix-multiprocessing"><span class="header-section-number">4</span> Matrix MultiProcessing</a>
  <ul>
  <li><a href="#basic-multiprocessing" id="toc-basic-multiprocessing" class="nav-link" data-scroll-target="#basic-multiprocessing"><span class="header-section-number">4.1</span> Basic (MultiProcessing)</a></li>
  <li><a href="#ci" id="toc-ci" class="nav-link" data-scroll-target="#ci"><span class="header-section-number">4.2</span> CI</a></li>
  </ul></li>
  <li><a href="#matrix-processing" id="toc-matrix-processing" class="nav-link" data-scroll-target="#matrix-processing"><span class="header-section-number">5</span> Matrix Processing</a>
  <ul>
  <li><a href="#annot.-columns" id="toc-annot.-columns" class="nav-link" data-scroll-target="#annot.-columns"><span class="header-section-number">5.1</span> Annot. columns</a></li>
  <li><a href="#annot.-rows" id="toc-annot.-rows" class="nav-link" data-scroll-target="#annot.-rows"><span class="header-section-number">5.2</span> Annot. rows</a></li>
  <li><a href="#basic-processing" id="toc-basic-processing" class="nav-link" data-scroll-target="#basic-processing"><span class="header-section-number">5.3</span> Basic (Processing)</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Activities</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Cox Lab </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 3, 2023</p>
    </div>
  </div>
  
    
  </div>
  


</header>

<!-- 
<ifauth @internal>//You are a user associated with MPIB, so you are presumably
using the internal version of Perseus, which has some features not available
to the general public. These internal features are visible when you view this page,
but not when an external user does so. They are marked with an asterisk (*).//</ifauth>

<ifauth @bioinformatics>//You are also associated with the bioinformatics group.
For you, developmental activities are also shown here and linked to whatever
documentation might be available, even though they are not available in the software.
They are marked with two asterisks (**).//</ifauth>
 -->
<p>Data processing in Perseus generally takes the form of loading a matrix of data and then applying a series of transformations to that matrix. These transformations are called “activities”. They are organized into a series of “Types”, and possibly a number of “Headings” under the Type.</p>
<p>You can search this page for the name of the activity of interest, or you can browse the Type or Heading of interest using the Table of Contents on the right.</p>
<section id="matrix-upload" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Matrix Upload</h1>
<ul>
<li><a href="docs/binaryupload.html">Binary upload</a></li>
<li><a href="docs/creategenelist.html">Create gene list</a></li>
<li><a href="docs/createrandommatrix.html">Create random matrix</a></li>
<li><a href="docs/genericmatrixupload.html">Generic matrix upload</a></li>
<li><a href="docs/ngsupload.html">Next generation sequencing data upload</a></li>
<li><a href="docs/unstructuredtxtupload.html">Raw upload</a></li>
</ul>
<!-- 
* [Affymetrix CGH](AffymetrixCghDataUpload) [[perseus:bioinformatics:activities:MatrixUpload:|]]
* [Agilent](AgilentDataUpload) [[perseus:bioinformatics:activities:MatrixUpload:|]]
* [SBML to matrix](Sbml2Matrix) [[perseus:bioinformatics:activities:MatrixUpload:|]]
 -->
</section>
<section id="matrix-export" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Matrix Export</h1>
<ul>
<li><a href="docs/tabseparatedexport.html">Generic matrix export</a></li>
</ul>
<!-- 
* [.arff export](ArffWriter) [[perseus:bioinformatics:activities:MatrixExport:|]]
 -->
</section>
<section id="matrix-analysis" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Matrix Analysis</h1>
<!--
## ARACNe (Analysis)


* [CoExpression clustering](ARACNeClustering) [[perseus:internal:activities:MatrixAnalysis:ARACNe:|]]


<!-- 
## Clustering

* [fastcluster](FastCluster) [[perseus:bioinformatics:activities:MatrixAnalysis:Clustering:|]]
 -->
<section id="clusteringpca" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="clusteringpca"><span class="header-section-number">3.1</span> Clustering/PCA</h2>
<ul>
<li><a href="docs/hierarchicalcluster.html">Hierarchical clustering</a></li>
<li><a href="docs/principalcomponentanalysis.html">Principal component analysis</a></li>
</ul>
<!-- 
* [Biclustering](Biclustering) [[perseus:bioinformatics:activities:MatrixAnalysis:ClusteringPCA:|]]
* [Factor analysis](FactorAnalysis) [[perseus:bioinformatics:activities:MatrixAnalysis:ClusteringPCA:|]]
* [Independent component analysis](IndependentComponentAnalysis) [[perseus:bioinformatics:activities:MatrixAnalysis:ClusteringPCA:|]]
 -->
<!-- 
# Co-expression
</ifauth>

* [Pick soft threshold](PickSoftThreshold) [[perseus:bioinformatics:activities:MatrixAnalysis:Co-expression]]
 -->
</section>
<section id="misc.-analysis" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="misc.-analysis"><span class="header-section-number">3.2</span> Misc. (Analysis)</h2>
<ul>
<li><a href="docs/numericvenndiagram.html">Numeric venn diagram</a></li>
<li><a href="docs/selectrowsmanually.html">Select rows manually</a></li>
<li><a href="docs/sequencelogo.html">Sequence logos</a></li>
<li><a href="docs/volcanoplotanalysis.html">Volcano plot</a></li>
</ul>
<!-- 
* [Edit matrix](EditMatrix) [[perseus:bioinformatics:activities:MatrixAnalysis:Misc:|]]
* [Hawaii plot (multi-volcano)](MultiVolcanoAnalysis) [[perseus:internal:activities:MatrixAnalysis:Misc:|]]
* [Sequence correlation](SequenceCorrelation) [[perseus:bioinformatics:activities:MatrixAnalysis:Misc:|]]
 -->
</section>
<section id="visualization" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="visualization"><span class="header-section-number">3.3</span> Visualization</h2>
<ul>
<li><a href="docs/scatterplot3danalysis.html">3D plot</a></li>
<li><a href="docs/histogram.html">Histogram</a></li>
<li><a href="docs/multiscatteranalysis.html">Multi scatter plot</a></li>
<li><a href="ProfilePlot">Profile plot</a> [[perseus:user:activities:MatrixAnalysis:Visualization:|]]\</li>
<li><a href="docs/scatterplotanalysis.html">Scatter plot</a> [[perseus:user:activities:MatrixAnalysis:Visualization:|]]\</li>
</ul>
<!-- 
* [Tree map](TreeMapAnalysis) [[perseus:bioinformatics:activities:MatrixAnalysis:Visualization:|]]**\\</ifauth>
* [Box plot](BoxPlotAnalysis) [[perseus:bioinformatics:activities:MatrixAnalysis:Visualization:|]]**\\</ifauth>
* [Charts](ChartAnalysis) [[perseus:bioinformatics:activities:MatrixAnalysis:Visualization:|]]**\\</ifauth>
 -->
</section>
</section>
<section id="matrix-multiprocessing" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Matrix MultiProcessing</h1>
<section id="basic-multiprocessing" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="basic-multiprocessing"><span class="header-section-number">4.1</span> Basic (MultiProcessing)</h2>
<p><em><a href="docs/matchingcolumnsbyname.html">Matching columns by name</a> </em><a href="docs/matchingrowsbyname.html">Matching rows by name</a> *<a href="docs/replacestrings.html">Replace strings</a></p>
<!-- 
*[Venn diagram](VennMultiProcessing) [[perseus:internal:activities:MatrixMultiProcessing:Basic:|]]
-->
</section>
<section id="ci" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="ci"><span class="header-section-number">4.2</span> CI</h2>
<ul>
<li><a href="docs/assertmatrixequal.html">Assert matrix equals</a></li>
</ul>
<!--  
## Misc. (MultiProcessing)

*[Match by chromosomal location](MatchingByGenomePosition) [[perseus:bioinformatics:activities:MatrixMultiProcessing:Misc:|]]
-->
</section>
</section>
<section id="matrix-processing" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Matrix Processing</h1>
<!-- 
## AA sequence

*[GRAVY](GravyValue) [[perseus:internal:activities:MatrixProcessing:AAsequence:|]]
*[Load sequences from fasta](LoadSequencesFromFasta) [[perseus:internal:activities:MatrixProcessing:AAsequence:|]]

## ARACNe (Processing)

*[Create adjacency matrix](ARACNeAdjacencyMatrix) [[perseus:internal:activities:MatrixProcessing:ARACNe:|]]

## Add content

* [IntAct neighborhood](IntActAnnotation) [[perseus:bioinformatics:activities:MatrixProcessing:Addcontent:|]]
* [Mouse phenotype](MousePhenotype) [[perseus:bioinformatics:activities:MatrixProcessing:Addcontent:|]]
* [Orthologs](AddOrthologs) [[perseus:bioinformatics:activities:MatrixProcessing:Addcontent:|]]
* [miRNA targets](MirnaTarget) [[perseus:bioinformatics:activities:MatrixProcessing:Addcontent:|]]
-->
<section id="annot.-columns" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="annot.-columns"><span class="header-section-number">5.1</span> Annot. columns</h2>
<ul>
<li><a href="docs/onedimannotationenrichment.html">1D annotation enrichment</a></li>
<li><a href="docs/twodimannotationenrichment.html">2D annotation enrichment</a></li>
<li><a href="docs/addannotationtomatrix.html">Add annotation</a></li>
<li><a href="docs/addannotationtomatrix.html">Annotation matrix</a></li>
<li><a href="docs/averagecategories.html">Average categories</a></li>
<li><a href="docs/categorycounting.html">Category counting</a></li>
<li><a href="docs/fisherexacttestproces.html">Fisher exact test</a></li>
<li><a href="docs/backtobaseidentifiers.html">To base identifiers</a></li>
</ul>
<!-- 
* [Add search database](AddSourceDatabase) [[perseus:internal:activities:MatrixProcessing:Annotcolumns:|]]
* [ComplexCorrelation](ComplexCorrelation) [[perseus:bioinformatics:activities:MatrixProcessing:Annotcolumns:]]
* [Count protein sequence features](AddProteinSequenceCounts) [[perseus:internal:activities:MatrixProcessing:Annotcolumns:|]]
-->
</section>
<section id="annot.-rows" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="annot.-rows"><span class="header-section-number">5.2</span> Annot. rows</h2>
<ul>
<li><a href="docs/averagegroups.html">Average groups</a></li>
<li><a href="docs/managecategoricalannotrow.html">Categorical annotation rows</a></li>
<li><a href="docs/jointermsincategoricalrow.html">Join terms in categorical row</a></li>
<li><a href="docs/managenumericalannotrow.html">Numerical annotation rows</a></li>
</ul>
<!-- 
* [Regex category row](RegexCategoryRow) [[perseus:internal:activities:MatrixProcessing:Annotrows:|]]
* [Regroup](Regroup) [[perseus:bioinformatics:activities:MatrixProcessing:Annotrows:|]]
-->
</section>
<section id="basic-processing" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="basic-processing"><span class="header-section-number">5.3</span> Basic (Processing)</h2>
<ul>
<li><a href="docs/addnoise.html">Add noise</a></li>
<li><a href="docs/cloneprocessing.html">Clone</a></li>
<li><a href="docs/columncorrelations.html">Column correlation</a></li>
<li><a href="docs/combinemaincolumns.html">Combine main columns</a></li>
<li><a href="combinebyidentifiersprocessing">Combine rows by identifiers</a> [[perseus:user:activities:MatrixProcessing:Basic:|]]\</li>
<li><a href="DensityEstimationProcessing">Density estimation</a> [[perseus:user:activities:MatrixProcessing:Basic:|]]\</li>
<li><a href="PerformanceCurves">Performance curves</a> [[perseus:user:activities:MatrixProcessing:Basic:|]]\</li>
<li><a href="Quantiles">Quantiles</a> [[perseus:user:activities:MatrixProcessing:Basic:|]]\</li>
<li><a href="RowCorrelations">Row correlation</a> [[perseus:user:activities:MatrixProcessing:Basic:|]]\</li>
<li><a href="SummaryStatisticsColumns">Summary statistics (columns)</a> [[perseus:user:activities:MatrixProcessing:Basic:|]]\</li>
<li><a href="SummaryStatisticsRows">Summary statistics (rows)</a> [[perseus:user:activities:MatrixProcessing:Basic:|]]\</li>
<li><a href="TransformationProc">Transform</a> [[perseus:user:activities:MatrixProcessing:Basic:|]]\</li>
</ul>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Basic:CreateNumericalColumn|Create numerical column]]<strong>\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Basic:CreateNumericalRow|Create numerical row]]</strong>\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Basic:FastPearsonCorrelations|Fast Pearson column correlation]]<strong>\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Basic:ParameterDependentCorrelation|Parameter dependent correlation]]</strong>\</p>
<p>==== Clustering ====</p>
<p>• [[perseus:user:activities:MatrixProcessing:Clustering:GenericClusteringProc|Generic clustering]]\</p>
<p>&lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Clustering:KernelPcaProc|Kernel PCA]]<em>\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Clustering:KernelCanonicalCorrelationProc|Kernel canonical correlation]]</em>\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Clustering:MotifClusteringProc|Motif clustering]]<em>\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Clustering:PeptideClusteringProc|Peptide clustering]]</em>\</p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;==== Co-expression ==== </p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Co-expression:MatrixModuleEigengene|Module Eigengenes]]**\</p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;==== Filter ==== </p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Filter:FilterLabels|Filter labels]]**\</p>
<p>==== Filter columns ====</p>
<p>• [[perseus:user:activities:MatrixProcessing:Filtercolumns:FilterCategoricalRow|Filter columns based on categorical row]]\ • [[perseus:user:activities:MatrixProcessing:Filtercolumns:FilterValidValuesColumns|Filter columns based on valid values]]\</p>
<p>==== Filter rows ====</p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Filterrows:GroupedValidValueFilter|Filter grouped rows by valid values – Jan]]**\ • [[perseus:user:activities:MatrixProcessing:Filterrows:FilterCategoricalColumn|Filter rows based on categorical column]]\ • [[perseus:user:activities:MatrixProcessing:Filterrows:FilterNumericalColumn|Filter rows based on numerical/main column]]\ • [[perseus:user:activities:MatrixProcessing:Filterrows:FilterRandomRows|Filter rows based on random sampling]]\ • [[perseus:user:activities:MatrixProcessing:Filterrows:FilterTextualColumn|Filter rows based on text column]]\ • [[perseus:user:activities:MatrixProcessing:Filterrows:FilterValidValuesRows|Filter rows based on valid values]]\</p>
<p>&lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;==== Genome ====</p>
<p>&lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Genome:AverageAlongChromosomes|Average along chromosomes]]<em>\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Genome:AddGenomePosition|Create genome position]]</em>\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Genome:GenomeAnalysis|Genome profiling]]*\</p>
<p>==== Imputation ====</p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Imputation:BpcaImputation|BPCA imputation]]**\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Imputation:KnnImputation|KNN imputation]]<em>\ • [[perseus:user:activities:MatrixProcessing:Imputation:ReplaceImputedByNan|Replace imputed values by NaN]]\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Imputation:ImputeSilacDoublets|Replace missing SILAC pair ratios]]<strong>\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Imputation:ImputeSilacTriplets|Replace missing SILAC triplet ratios]]</strong>\ • [[perseus:user:activities:MatrixProcessing:Imputation:ReplaceMissingByConstant|Replace missing values by constant]]\ • [[perseus:user:activities:MatrixProcessing:Imputation:ReplaceMissingFromGaussian|Replace missing values from normal distribution]]\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Imputation:RowAverageImputation|Row average imputation]]</em>\</p>
<p>&lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;==== Interaction ====</p>
<p>&lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Interaction:InteractionDbAnnotation|Annotate from category row]]*\</p>
<p>==== Learning ====</p>
<p>• [[perseus:user:activities:MatrixProcessing:Learning:ClassificationProcessing|Classification (cross-validation and prediction)]]\ • [[perseus:user:activities:MatrixProcessing:Learning:ClassificationFeatureSelection|Classification feature optimization]]\ • [[perseus:user:activities:MatrixProcessing:Learning:ClassificationParameterOptimization|Classification parameter optimization]]\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Learning:RegressionProcessing|Regression (cross-validation and prediction)]]<em>\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Learning:RegressionFeatureSelection|Regression feature optimization]]</em>\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Learning:RegressionParameterOptimization|Regression parameter optimization]]*\</p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;==== Misc. (Processing) ==== </p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Misc:CreateGeneLists|Create gene lists]]<strong>\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Misc:AssemblePeptides|Peptide assembly]]</strong>\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Misc:SmallMolMasses|Small mol masses]]**\</p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;==== Modeling ==== </p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Modeling:SimmuneSimulationResultToMqOutput|Simulation results to MQ output]]**\</p>
<p>==== Modifications ====</p>
<p>• [[perseus:user:activities:MatrixProcessing:Modifications:AddKnownSites|Add known sites]]\ • [[perseus:user:activities:MatrixProcessing:Modifications:AddLinearMotifs|Add linear motifs]]\ • [[perseus:user:activities:MatrixProcessing:Modifications:AddModificationCounts|Add modification counts]]\ • [[perseus:user:activities:MatrixProcessing:Modifications:AddRegulatorySites|Add regulatory sites]]\ • [[perseus:user:activities:MatrixProcessing:Modifications:AddSequenceFeatures|Add sequence features]]\ • [[perseus:user:activities:MatrixProcessing:Modifications:ExpandSiteTable|Expand site table]]\ • [[perseus:user:activities:MatrixProcessing:Modifications:KinaseSubstrateRelations|Kinase-substrate relations]]\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Modifications:SequenceWindowToVector|Sequence window conversion]]**\ • [[perseus:user:activities:MatrixProcessing:Modifications:ShortenMotifs|Shorten motif length]]\</p>
<p>==== Normalization ====</p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Normalization:BackgroundSubtraction|Background subtraction]]**\ • [[perseus:user:activities:MatrixProcessing:Normalization:ClusterNorm|Cluster normalization]]\ • [[perseus:user:activities:MatrixProcessing:Normalization:Divide|Divide]]\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Normalization:IntensityDependentNormalization2|Intensity-dependent normalization]]*\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Normalization:IntensityDependentNormalization|LOWESS normalization]]<strong>\ • [[perseus:user:activities:MatrixProcessing:Normalization:NormalizeByColumn|Modify by column]]\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Normalization:NormalizeByGroup|Normalize by group – Jan]]</strong>\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Normalization:OptimalRotation|Optimal rotation]]**\ • [[perseus:user:activities:MatrixProcessing:Normalization:Rank|Rank]]\ • [[perseus:user:activities:MatrixProcessing:Normalization:ScaleToInterval|Scale to interval]]\ • [[perseus:user:activities:MatrixProcessing:Normalization:Subtract|Subtract]]\ • [[perseus:user:activities:MatrixProcessing:Normalization:SubtractRowCluster|Subtract row cluster]]\ • [[perseus:user:activities:MatrixProcessing:Normalization:UnZScore|Un-Z-score]]\ • [[perseus:user:activities:MatrixProcessing:Normalization:UnitVector|Unit vectors]]\ • [[perseus:user:activities:MatrixProcessing:Normalization:WidthAdjustment|Width adjustment]]\ • [[perseus:user:activities:MatrixProcessing:Normalization:ZScore|Z-score]]\</p>
<p>==== Outliers ====</p>
<p>&lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Outliers:ForwardReverseAnalysis|Forward-reverse analysis]]**\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Outliers:MultiDimSignificance|Multidimensional significance]]*\ • [[perseus:user:activities:MatrixProcessing:Outliers:SignificanceA|Significance A]]\ • [[perseus:user:activities:MatrixProcessing:Outliers:SignificanceB|Significance B]]\</p>
<p>==== Quality ====</p>
<p>• [[perseus:user:activities:MatrixProcessing:Quality:ConvertToNaN|Convert to NaN]]\ • [[perseus:user:activities:MatrixProcessing:Quality:CreateQualityMatrix|Create quality matrix]]\ • [[perseus:user:activities:MatrixProcessing:Quality:FilterQuality|Filter quality]]\</p>
<p>==== Rearrange ====</p>
<p>• [[perseus:user:activities:MatrixProcessing:Rearrange:ChangeColumnType|Change column type]]\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Rearrange:CollapseEvidenceFile|Collapse MaxQuant evidence File]]<em>\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Rearrange:ColumnGroupingToMultipleRows|Column grouping to multiple rows]]</em>\ • [[perseus:user:activities:MatrixProcessing:Rearrange:CombineAnnotations|Combine annotations]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:CombineCategoricalColumns|Combine categorical columns]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:ConvertMultiNumeric|Convert multi-numeric column]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:DeHyphenateIds|De-hyphenate ids]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:DuplicateColumns|Duplicate columns]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:ExpandMultiNumeric|Expand multi-numeric and text columns]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:FillCategoricalColumn|Fill categorical columns]]\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Rearrange:MapColumnsToUniprot|Map columns to Uniprot – Jan]]**\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Rearrange:MatrixToList|Matrix to List]]*\ • [[perseus:user:activities:MatrixProcessing:Rearrange:ProcessTextColumns|Process text column]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:RemoveEmptyColumns|Remove empty columns]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:RenameColumns|Rename columns]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:RenameColumnsRegexp|Rename columns [reg. ex.]]]\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Rearrange:RenameColumnsFromFile|Rename columns from file – Jan]]<strong>\ • [[perseus:user:activities:MatrixProcessing:Rearrange:ReorderColumnsByNumAnnotationRow|Reorder columns by numerical annotation row]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:ReorderRemoveAnnotRows|Reorder/remove annotation rows]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:ReorderRemoveColumns|Reorder/remove columns]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:SearchTextColumns|Search text column]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:SortByColumn|Sort by column]]\ &lt;ifauth <span class="citation" data-cites="bioinformatics">(<a href="#ref-bioinformatics" role="doc-biblioref"><strong>bioinformatics?</strong></a>)</span>&gt;• [[perseus:bioinformatics:activities:MatrixProcessing:Rearrange:SortColumns|Sort columns – Jan]]</strong>\ • [[perseus:user:activities:MatrixProcessing:Rearrange:Transpose|Transpose]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:UniqueRows|Unique rows]]\ • [[perseus:user:activities:MatrixProcessing:Rearrange:UniqueValues|Unique values]]\</p>
<p>==== Tests ====</p>
<p>&lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Tests:CorrelationTests|Correlation tests]]<em>\ &lt;ifauth <span class="citation" data-cites="internal">(<a href="#ref-internal" role="doc-biblioref"><strong>internal?</strong></a>)</span>&gt;• [[perseus:internal:activities:MatrixProcessing:Tests:FourWayAnova|Four-way ANOVA]]</em>\ • [[perseus:user:activities:MatrixProcessing:Tests:MultipleSampleTestProcessing|Multiple-sample tests]]\ • [[perseus:user:activities:MatrixProcessing:Tests:OneSampleTestProcessing|One-sample tests]]\ • [[perseus:user:activities:MatrixProcessing:Tests:PostHocTests|Post hoc tests]]\ • [[perseus:user:activities:MatrixProcessing:Tests:ThreeWayAnova|Three-way ANOVA]]\ • [[perseus:user:activities:MatrixProcessing:Tests:TwoSampleTestProcessing|Two-sample tests]]\ • [[perseus:user:activities:MatrixProcessing:Tests:TwoWayAnova|Two-way ANOVA]]\</p>
<p>==== Time series ====</p>
<p>• [[perseus:user:activities:MatrixProcessing:Timeseries:CyclicAnnotation|Cyclic annotation enrichment]]\ • [[perseus:user:activities:MatrixProcessing:Timeseries:PeriodicityAnalysis|Periodicity analysis]]\ • [[perseus:user:activities:MatrixProcessing:Timeseries:Periodogram|Periodogram]]\ • [[perseus:user:activities:MatrixProcessing:Timeseries:TimeSeriesOrdering|Time series ordering]]\</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        for (let i = 0; i < 2; i++) {
          container.appendChild(note.children[i].cloneNode(true));
        }
        return container.innerHTML
      } else {
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      return note.innerHTML;
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>© 2023, created by and maintained by Cox lab</p>
</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>




</body></html>